docs/nnx/bridge_guide.md

```diff
@@ -0,0 +1,544 @@
+
+Another difference is that NNX modules are stateful, so they can track and update the RNG keys within themselves. This means a bit difference in both `ToNNX` and `ToLinen`:
+
+### Linen to NNX
+
+If you convert a Linen module to NNX, you enjoy the stateful benefit and don't need to pass in extra RNG keys on every module call. You can use always `nnx.reseed` to reset the RNG state within.
+
+
+```python
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.ToNNX(nn.Dropout(rate=0.5, deterministic=False), rngs=nnx.Rngs(dropout=0))
+bridge.lazy_init(model, x)       # We don't really need this b/c no extra params were created here,
+                                 # but it's a good practice to always add this line.
+y1, y2 = model(x), model(x)
+assert not jnp.allclose(y1, y2)  # Two runs yield different outputs!
+
+# Reset the dropout RNG seed, so that next model run will be the same as the first.
+nnx.reseed(model, dropout=0)
+assert jnp.allclose(y1, model(x))
+```
+
+### NNX to Linen
+
+If you convert an NNX module to Linen, the underlying NNX module's RNG states will still be part of the top-level `variables`. On the other hand, Linen `apply()` call accepts different RNG keys on each call, which resets the internal Linen environment and allow different random data to be generated.
+
+Now, it really depends on whether your underlying NNX module generates new random data from its RNG state, or from the passed-in argument. Fortunately, `nnx.Dropout` supports both - using passed-in keys if there is any, and use its own RNG state if not.
+
+And this leaves you with two style options of handling the RNG keys: 
+
+* The NNX style (recommended): Let the underlying NNX state manage the RNG keys, no need to pass in extra keys in `apply()`. This means a few more lines to mutate the `variables` for every apply call, but things will look easier once your whole model no longer needs `ToLinen`.
+
+* The Linen style: Just pass different RNG keys for every `apply()` call.
+
+
+```python
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.to_linen(nnx.Dropout, rate=0.5)
+variables = model.init({'dropout': jax.random.key(0)}, x)
+
+# The NNX RNG state was stored inside `variables`
+print('The RNG key in state:', variables['RngKey']['rngs']['dropout']['key'].value)
+print('How many times we have created new RNG keys from it:', variables['RngCount']['rngs']['dropout']['count'].value)
+
+# NNX style: Must set `RngCount` as mutable and update the variables after every `apply`
+y1, updates = model.apply(variables, x, mutable=['RngCount'])
+variables |= updates
+y2, updates = model.apply(variables, x, mutable=['RngCount'])
+assert not jnp.allclose(y1, y2)  # Every call yields different output!
+
+# Linen style: Just pass different RNG keys for every `apply()` call.
+y3 = model.apply(variables, x, rngs={'dropout': jax.random.key(1)})
+y4 = model.apply(variables, x, rngs={'dropout': jax.random.key(2)})
```

Comment:
could be interesting to do the opposite and use the same key to show that this overrides the keys and yields the same output

Reply:
Good idea! done.